### PR TITLE
EOS-28388: fetch data from bytecount btree

### PR DESCRIPTION
**Pull request by AbhishekSahaSeagate**
sss foms now make m0_cob_bc_entries_dump() calls to populate
reply fop key record buffers with bytecount data from bytecount btree.

Changed implementation of spiel API m0_spiel_proc_counters_fetch() to
accept data from btree instead of hardcoded data.

Signed-off-by: Abhishek Saha <abhishek.saha@seagate.com>

# Problem Statement
- Problem statement

# Design
-  For Bug, Describe the fix here.
-  For Feature, Post the link for design

# Coding
   Checklist for Author
-  [ ] Coding conventions are followed and code is consistent

# Testing 
  Checklist for Author
- [ ] Unit and System Tests are added
- [ ] Test Cases cover Happy Path, Non-Happy Path and Scalability
- [ ] Testing was performed with RPM

# Impact Analysis
  Checklist for Author/Reviewer/GateKeeper
- [ ] Interface change (if any) are documented
- [ ] Side effects on other features (deployment/upgrade)
- [ ] Dependencies on other component(s)

# Review Checklist 
  Checklist for Author
- [ ] JIRA number/GitHub Issue added to PR
- [ ] PR is self reviewed
- [ ] Jira and state/status is updated and JIRA is updated with PR link
- [ ] Check if the description is clear and explained

# Documentation
  Checklist for Author
- [ ] Changes done to WIKI / Confluence page / Quick Start Guide

### CODE REVIEW

File: cob/cob.c [hunk 1]
```diff
@@ -500,66 +494,59 @@ M0_INTERNAL void m0_cob_bc_iterator_fini(struct m0_cob_bc_iterator *it)
...
-	m0_buf_alloc(out_keys, sizeof(struct m0_cob_bckey) * (*out_count));
-	m0_buf_alloc(out_recs, sizeof(struct m0_cob_bcrec) * (*out_count));
-	key_cursor = out_keys->b_addr;
-	rec_cursor = out_recs->b_addr;
+	M0_ALLOC_PTR(*out_keys);
+	M0_ALLOC_PTR(*out_recs);
 
-	m0_list_for_each(&entry_list, list_pos) {
-		struct bc_entry *entry;
+	m0_buf_alloc(*out_keys, sizeof(struct m0_cob_bckey) * (*out_count));
```
Comment (madhavemuri) [inline cob/cob.c:530]: Check rc here.

File: spiel/cmd.c [hunk 2]
```diff
@@ -2059,8 +2060,14 @@ static void spiel_process_counter_replied_ast(struct m0_sm_group *grp,
 		goto leave;
 	rep = spiel_process_reply_data(&proc->sci_fop);
 
-	proc->sci_data.pd_bckey = *((struct m0_spiel_bckey *)rep->sspr_bckey.b_addr);
-	proc->sci_data.pd_bcrec = *((struct m0_spiel_bcrec *)rep->sspr_bcrec.b_addr);
+	m0_buf_alloc(&proc->sci_data.pd_key, rep->sspr_bckey.b_nob);
```
Comment (madhavemuri) [inline spiel/cmd.c:2063]: Check rc here.

File: sss/process_foms.c [hunk 3]
```diff
@@ -380,32 +381,47 @@ static int ss_process_stats(struct m0_reqh           *reqh,
...
+		return 0;
+	}
 
-	m0_buf_new_aligned(&rep->sspr_bckey, &key, sizeof(key), 0);
-	m0_buf_new_aligned(&rep->sspr_bcrec, &rec, sizeof(rec), 0);
+	rc = m0_cob_bc_entries_dump(cdom, &key_buf, &rec_buf, &rep->sspr_kv_count);
+	if (rc == 0) {
+		m0_buf_init(&rep->sspr_bckey, key_buf->b_addr, key_buf->b_nob);
+		m0_buf_init(&rep->sspr_bcrec, rec_buf->b_addr, rec_buf->b_nob);
+	}
```
Comment (madhavemuri) [inline sss/process_foms.c:410]: Check where these buffers are getting freed.

Comment (AbhishekSahaSeagate) [reply sss/process_foms.c:410]: It is getting freed in m0_ss_process_rep_fop_release(). This change was added in #1446
sss/process_fops.h

**Review by madhavemuri — APPROVED** [review]

**Pull request merged**